python/Sign_Extender.py

```diff
@@ -4,12 +4,12 @@
 def Sign_Extender(clk, instruction, output):
     #Instruction should be a 16-bit intbv
     #output is accurate 32-bit intbv.
-    output = intbv(max = 32)
+    output.Next = intbv(min = 32, max = 32)
     
     @alway(clk.posedge)
     def execute():
         #Insert 16 to 32-bit conversion loginc
-
-        return
+        output[31:16].Next = instruciton[15] #make the first new bits match the first bit
+        return output.Next
```

Comment:
Return values in the inner myhdl functions are useless. It's impossible to retrieve the value from them due to the round about way the function is called. Output is done through the passed output parameter which is already implemented in the file!